### ISSUE TITLE
Qubes Release Should Always Contain Functional Tor/Whonix

### ISSUE THREAD

**Issue by straussmaus** [QubesOS/qubes-issues#7160]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### The problem you're addressing (if any)

Currently, release 4.0.4 contains sys-whonix-15.  But Whonix 15 is no longer supported.  Therefore there is no usable Whonix in the most up-to-date Qubes release.

### The solution you'd like

I recommend a policy that all Qubes releases be held until it is verified that the current whonix template has at least X months of support left.  (Either 3 months or whenever the next Qubes release is due?)

### The value to a user, and who that user might be

It is valuable for all of the reasons that Whonix is valuable.  And it is always valuable to only include features that are functional at the time they are provided.

**Comment by fireneat** [comment]
There will probably soon be a 4.0.5 release which will have the updated templates #7161

**Comment by andrewdavidwong** [comment]
**Note for devs:**

I think the best way to understand this issue is as a request to consider modifying our _release policy or process_. In other words, I think it would be a mistake to think that shipping any particular technical deliverable (such as a new `4.X.Y` ISO that contains up-to-date Whonix templates) would speak directly to this issue, since it could easily recur again down the line. Rather, a decision needs to be made whether to modify the release process or policy in some way that prevents the undesired outcome from recurring.

Please do not misinterpret my statements here as agreeing or disagreeing with the request. I'm simply explaining how best to _understand_ the request. It's up to you, devs, to decide whether you agree with it and are willing and able to implement it.

Now, a specific observation:
> I recommend a policy that all Qubes releases be held until it is verified that the current whonix template has at least X months of support left. (Either 3 months or whenever the next Qubes release is due?)

Since we often do not know, in advance, when the next Qubes release is "due," this policy would not, by itself, prevent the undesired outcome. Imagine the following scenario:

- Qubes Release `X.Y.Z` is released, which includes Whonix templates with 3 months of support remaining.
- Due to various delays, no further Qubes versions are released for 5 months.

In this scenario, the policy was obeyed, and yet the undesired outcome still occurred. Since we do not presently have guaranteed new releases at predictable intervals, the policy would have to be the other way around (_requiring_ rather than _blocking_ the release of a new ISO) in order to achieve the desired effect.

**Comment by andrewdavidwong** [comment]
Closely related to: https://github.com/QubesOS/qubes-issues/issues/6515

**Comment by straussmaus** [comment]
> In this scenario, the policy was obeyed, and yet the undesired outcome still occurred.

True.  Although at least in that case a user could think "I'll wait for the next release" and be certain that that *would* fix the problem.  What ships will be fully functional when it is released.  At present, a user could go through the significant difficulty of downloading a brand new release (it takes me about three days), installing, configuring, and then finding that he has a brick even for something released just recently.

Anyway yes you understand the request correctly.  Thank you for considering.
